Branch-SDK/BranchOpenRequest.m

```diff
@@ -85,6 +86,11 @@ - (void)makeRequest:(BNCServerInterface *)serverInterface key:(NSString *)key ca
                     forKey:BRANCH_REQUEST_KEY_SEARCH_AD
                     onDict:params];
     }
+    
+    NSDictionary *partnerParameters = [[BNCPartnerParameters shared] parameterJson];
+    if (partnerParameters.count > 0) {
+        [self safeSetValue:partnerParameters forKey:BRANCH_REQUEST_KEY_PARTNER_PARAMETERS onDict:params];
```

Comment:
Just a general thought. I know this method has been around for a while, but it seems like this would make more sense as a category on NSDictionary, e.g. `[params safeSetValue:partnerParameters forKey:BRANCH_REQUEST_KEY_PARTNER_PARAMETERS]`. I think when using modern syntax though, you may be able to assign nil, i.e. `params[BRANCH_REQUEST_KEY_PARTNER_PARAMETERS] = partnerParameters // ok if it's nil`. But that's more a question of general refactoring.

Reply:
Yea I agree.  I think some of these old methods could use modernization.